### PR TITLE
Custom template for config resource

### PR DESCRIPTION
**Pull request by limitusus**
# Description

Enable to specify template cookbook for config resource.

Specifically, I want to modify `ErrorLog` directive to pipe, not file.
Current `apache2.conf.erb` can't express such case, so I thought I should prepare an original template for that in my wrapper cookbook, but`apache2_config` resource has `cookbook` with 'apache2' hard-coded.
I think there is no reason not to let it configurable like `apache2_conf`, `apache_default_site` and so on.

## Issues Resolved

N/A

## Check List

- [x] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@limitusus can you please add a changelog entry and maybe some new tests?

**Comment by limitusus** [comment]
Added unit test. Is that enough or the same for integration test required?

**Review by ramereth — APPROVED** [review]
This LGTM as-is.

File: spec/resources/config_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe 'apache2_config' do
+  step_into :apache2_config
+  platform 'ubuntu'
+
+  context 'install config template from external' do
+    recipe do
+      service 'apache2' do
+        extend Apache2::Cookbook::Helpers
```
Comment (ramereth) [inline spec/resources/config_spec.rb:10]: Now that I merged #705, can you please remove this line?

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please resolve the conflicts?

**Review by ramereth — DISMISSED** [review]

**Review by ramereth — APPROVED** [review]

**Pull request merged**